pkg/test/dnszone/dnszone.go

```diff
@@ -115,3 +116,12 @@ func WithZone(zone string) Option {
 		dnsZone.Spec.Zone = zone
 	}
 }
+
+func WithGCPPlatform() Option {
+	return func(dnsZone *hivev1.DNSZone) {
+		dnsZone.Spec.GCP = &hivev1.GCPDNSZoneSpec{}
+		dnsZone.Status.GCP = &hivev1.GCPDNSZoneStatus{
+			ZoneName: pointer.StringPtr("testZoneName"),
```

Comment:
Pass "testZoneName" in as a parameter to `WithGCPPlatform` rather than assuming that the tests using this want the zone name to the "testZoneName".